### PR TITLE
Add anonymous reporting and remove app insights telemetry

### PR DESCRIPTION
**Pull request by jcolladokuri**
Instead of using Application Insights for reporting exceptions and events, this PR removes that and uses grafana's `usePluginInteractionReporter` which allows for anonymous reporting of different events.

### CODE REVIEW

File: src/components/SceneApp/AKSAppScene.tsx [hunk 1]
```diff
@@ -3,10 +3,14 @@ import { useSceneApp } from "@grafana/scenes";
 import { getSceneApp } from "./SceneApp";
 import { PluginPropsContext } from "utils/utils.plugin";
 import { AKSPluginProps } from "types";
+import { usePluginInteractionReporter } from "@grafana/runtime";
+import { TelemetryClient } from "telemetry/telemetry";
 
 export function AKSAppScene() {
     const { configState, setConfigState } = useContext(PluginPropsContext) as AKSPluginProps;
-    const scene = useSceneApp(() => getSceneApp(configState, setConfigState));
+    const report = usePluginInteractionReporter();
```
Comment (yves-chan) [inline src/components/SceneApp/AKSAppScene.tsx:11]: how do we access this telemetry?

File: src/components/SceneApp/Queries/ClusterByWorkloadQueries.ts [hunk 2]
```diff
@@ -235,14 +235,12 @@ function GetIconsOnCells(dataFrames: DataFrame[]): DataFrame[] {
...
-      exception: e instanceof Error ? e : new Error(JSON.stringify(e)),
-      severityLevel: SeverityLevel.Error,
-      properties: {
-        reporter: "Scene.Main.WorkloadsScene",
-        action: "TransformTableData"
-      }
-    })
+    telemetryClient.reportException("grafana_plugin_geticonsoncells_failed", {
+      reporter: "Scene.Main.WorkloadsScene",
+      exception: "e instanceof Error ? e : new Error(JSON.stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Queries/ClusterByWorkloadQueries.ts:240]: stringified error conditional

File: src/components/SceneApp/Pages/ComputeResourcesDrilldown.ts [hunk 3]
```diff
@@ -208,15 +217,13 @@ function getComputeResourcesDrilldownScene() {
...
-                    properties: {
-                        reporter: "Scene.Drilldown.ComputeResources",
-                        referer: "Scene.Main.WorkloadsScene",
-                        action: "changePromVariableOnClusterChange"
-                    }
-                });
+                telemetryClient.reportException("grafana_plugin_promdsvarchange_failed", {
+                    reporter: "Scene.Drilldown.ComputeResources",
+                    refererer: "Scene.Main.WorkloadsScene",
+                    exception: "e instanceof Error ? e : new Error(stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Pages/ComputeResourcesDrilldown.ts:223]: this is in quotes as well

File: src/components/SceneApp/Pages/ComputeResourcesDrilldown.ts [hunk 4]
```diff
@@ -228,16 +235,14 @@ function getComputeResourcesDrilldownScene() {
...
-                    reporter: "Scene.Drilldown.ComputeResources",
-                    referer: "Scene.Main.WorkloadsScene",
-                    action: "createClusterMappings"
-                }
-            });
+            } catch (e) {
+              telemetryClient.reportException("grafana_plugin_createclustermappings_failed", {
+                  reporter: "Scene.Drilldown.ComputeResources",
+                  refererer: "Scene.Main.WorkloadsScene",
+                  exception: "e instanceof Error ? e : new Error(stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Pages/ComputeResourcesDrilldown.ts:242]: this one is stringified

File: src/components/SceneApp/Pages/Namespaces.ts [hunk 5]
```diff
@@ -79,13 +86,11 @@ export function getNamespacesScene(): SceneAppPage {
...
-          trackException({
-            exception: e instanceof Error ? e : new Error(stringify(e)),
-            severityLevel: SeverityLevel.Error,
-            properties: {
-              reporter: "Scene.Main.NamespacesScene",
-              action: "changePromVariableOnClusterChange"
-            }
+          telemetryClient.reportException("grafana_plugin_promdsvarchange_failed", {
+            reporter: reporter,
+            exception: "e instanceof Error ? e : new Error(stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Pages/Namespaces.ts:91]: this one also

File: src/components/SceneApp/Pages/Nodes.ts [hunk 6]
```diff
@@ -107,14 +113,12 @@ export function getOverviewByNodeScene(): SceneAppPage {
...
-                    exception: e instanceof Error ? e : new Error(stringify(e)),
-                    severityLevel: SeverityLevel.Error,
-                    properties: {
-                      reporter: "Scene.Main.NodesScene",
-                      action: "runQueriesOnClusterChange"
-                    }
-                });
+                telemetryClient.reportException("grafana_plugin_runqueries_failed", {
+                    reporter: reporter,
+                    exception: "e instanceof Error ? e : new Error(stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Pages/Nodes.ts:118]: stringified error conditional

File: src/components/SceneApp/Pages/PodWithLogsDrilldown.ts [hunk 7]
```diff
@@ -361,15 +368,13 @@ function getPodWithLogsDrilldownScene() {
...
-                        properties: {
-                            reporter: "Scene.Drilldown.PodWithLogsDrilldown",
-                            referer: "Scene.Drilldown.ComputeResourcesDrilldown",
-                            action: "changePromVariableOnClusterChange"
-                        }
-                    });
+                    telemetryClient.reportException("grafana_plugin_promdsvarchange_failed", {
+                        reporter: "Scene.Drilldown.PodWithLogs",
+                        refererer: "Scene.Drilldown.ComputeResources",
+                        exception: "e instanceof Error ? e : new Error(stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Pages/PodWithLogsDrilldown.ts:374]: same here

File: src/components/SceneApp/Pages/Workloads.ts [hunk 8]
```diff
@@ -96,13 +102,11 @@ export function getClusterByWorkloadScene() {
...
-          trackException({
-            exception: e instanceof Error ? e : new Error(stringify(e)),
-            severityLevel: SeverityLevel.Error,
-            properties: {
-              reporter: "Scene.Main.WorkloadsScene",
-              action: "changePromVariableOnClusterChange"
-            }
+          telemetryClient.reportException("grafana_plugin_promdsvarchange_failed", {
+            reporter: reporter,
+            exception: "e instanceof Error ? e : new Error(stringify(e))",
```
Comment (yves-chan) [inline src/components/SceneApp/Pages/Workloads.ts:107]: here also

Comment (jcolladokuri) [reply src/components/SceneApp/Pages/Workloads.ts]: whoops, good catch, stringified it for testing.. removed all of them

File: src/components/SceneApp/AKSAppScene.tsx (hunk 1 above)
Comment (jcolladokuri) [reply src/components/SceneApp/AKSAppScene.tsx:11]: Grafana shares it with us upon request!

**Review by yves-chan — APPROVED** [review]

**Pull request merged**